### DISCUSSION TITLE
Unexpectedly high memory usage on macOS

### DISCUSSION THREAD

**Discussion by champagnealexandre** [iterate-ch/cyberduck#16431] [Q&A]
**Describe the bug**
Cyberduck leaks memory on macOS Monterey. For example, right now it’s using 1.14GB of RAM memory. This happens even after restarts, etc.

**To Reproduce**
I have set-up bookmarks for sftp and Google Drive. Not sure what’s causing this exactly.

**Expected behavior**
Use less than 1GB of RAM ;-)

**Screenshots**
<img width="659" alt="image" src="https://user-images.githubusercontent.com/30211986/183636020-0439187c-7064-4904-abb4-a52869d6e6fb.png">

<img width="484" alt="image" src="https://user-images.githubusercontent.com/30211986/183636073-c8e85d35-a11c-49a5-b85a-d12be903c683.png">

**Desktop (please complete the following information):**
 - OS: macOS Monterey 
 - Version 12.4 (21F79)

**Log Files**
See this link: https://drive.google.com/file/d/1lhGzvrr3-me9uTyv7FxfRTKUhhamqkgx/view?usp=sharing

**Additional context**
I have tried to reach out for support (mountainduck.io) and I was told to use a RAM memory cleaner (!!!)

**Comment by dkocher** [comment]
> I have tried to reach out for support (mountainduck.io) and I was told to use a RAM memory cleaner (!!!)

This sure does not make sense. But we run a virtual machine on macOS that will only eventually release memory on garbage collection and not necessarily when no longer retained. Thus you might see a decrease in memory usage only when there is memory pressure. I will close this as we use this issue tracker only for Cyberduck. We will follow up in your support request.

Relates to #12695.

**Comment by champagnealexandre** [comment]
Thanks for the follow up @dkocher.

**Comment by movinimage** [comment]
I'm running Version 4.13.8 (21184)...a Beta I believe....I just noticed MD was using 8.1 GB of memory. Ha! Restarted it,

**Comment by anthony-kyle** [comment]
At least once a day I need to restart Mountain duck cause my machine suddenly runs out of memory. Using v4.16.3

**Comment by twilsonco** [comment]
I'm experiencing this too with v4.16.3 on macOS 15.0.1 on a m1 Mac Mini. Just restarted Mountain Duck after it was consuming 2.6GB with one Dropbox share mounted.
![mountain duck high memory](https://github.com/user-attachments/assets/3203a544-4834-41cf-9509-c81c3c0b397a)

**Reply by minwoo-yu** [reply]
Is this still the case? I'm using an M2 MAC MINI and I'm still experiencing similar symptoms. I have about 3 mounting points enabled and it takes about 3~4GB of memory :(

**Comment by brinscall** [comment]
Let me one up you. This is insane
<img width="326" alt="leak" src="https://github.com/user-attachments/assets/1fbf6b3b-f4e7-4278-a2c2-003d5e616150">

**Reply by dkocher** [reply]
This discussion is about [Mountain Duck](https://mountainduck.io/) while your screenshot shows Cyberduck with heavy memory usage. We are tracking this issue in #15510. Please comment in the ticket with details you have about your use case including the data set and protocol in use.

**Comment by dkocher** [comment]
Relates to [^1].

[^1]: #14894

**Comment by jvsteiner** [comment]
MountainDuck Version 4.17.2 (22563)  regular using 3-4 Gb on a Macbook pro M1 with 16 Gb running Sequoia 15.2 (24C101).  I restart the beast several times a day.  Does not seem tied to any particular activity.  I also have cyberduck installed, and given there is an issue there as well, and that the two programs work cooperatively, I would not be shocked to learn that they share a root cause.

**Reply by jvsteiner** [reply]
I sometimes have watched it in the Activity Monitor.  It seems to start growing in memory usage quite quickly, while running.  Then, every minute or so, it gets garbage collected, and the usage drops.  Except when it doesn't.  The effect is that during idle - ie. I am not using it for anything, no transfers, etc. - the memory usage pogos up and down between about 250-500Mb.  This is just after restarting it - I haven't "caught it in the act" of claiming more memory than that (up to 6-9 Gb in my experience), but the behavior I describe already seems like there is a leak - just that it might be conditionally getting GC'd.

**Comment by jvsteiner** [comment]
Is this being worked on, or is there a way to pin an old version that is known or suspected not to have this problem?  I use Mountain duck in my daily workflow, and its great apart from this, but its mostly unuseable for me

**Comment by MisterDenis** [comment]
Just to throw my hat in the ring, very similar issue on Windows as well.
